pom.xml

```diff
@@ -67,6 +67,12 @@
         <spring.version>2.0.5.RELEASE</spring.version>
         <powermock.version>1.7.3</powermock.version>
         <lombok.version>1.18.0</lombok.version>
+        <errorprone.version>2.3.4</errorprone.version>
```

Comment:
As discussed, let's split the PR in Sonar warnings fixes and plugins. We'll look into details about the benefits of these plugins. Thanks!